### DISCUSSION TITLE
Updating the Menu in the Tray

### DISCUSSION THREAD

**Discussion by bmcgonag** [neutralinojs/neutralinojs#1159] [Q&A]
I have a simple app, and I'm getting the tray icon set with no issues using the following:

```
let tray;

function setTray() {
    if(NL_MODE != "window") {
        // console.log("INFO: Tray menu is only available in the window mode.");
        return;
    }
    tray = {
        icon: "/resources/icons/wg-cli-icon.png",
        menuItems: [
            {id: "SHOW", text: "Show"},
            {id: "SEP", text: "-"},
            {id: "QUIT", text: "Quit"}
        ]
    };
    Neutralino.os.setTray(tray);
}

function onTrayMenuItemClicked(event) {
    switch(event.detail.id) { 
        case "SHOW":
            Neutralino.window.show();
            break;
        case "QUIT":
            Neutralino.app.exit();
            break;
    }
}
```

In a separate function triggered by use action in the main window, I'm trying to dynamically add an entry into the tray menu like this:

```
tray.menuItems.unshift({id: interface, text: interface}, {id: 'SEP', text: '-'});

Neutralino.os.setTray(tray);
```

When I take the action, I can console logout the new `tray.menuItems` array of objects, and everything is there as you'd expect, but the tray itself isn't updated.  

Any help on what I may be doing wrong would be greatly appreciated.

**Comment by bmcgonag** [comment]
Upddate: This does work, sometimes, but it's definitely inconsistent.  Not sure why it doesn't trigger the update every time.

**Comment by bmcgonag** [comment]
I'm still having issues with the tray being functional every time I run the application.  Is this normal?  Sometimes the tray works exactly as expected, but other times it show, but clicking on the options has no effect.  I'm a bit concerned about that.

**Comment by shalithasuranga** [comment]
Hey @bmcgonag thanks for posting this. The `setTray` function is implemented to support dynamic menu updates. I'll try to reproduce this issue and check whether there is a bug in the tray menu dependency. Thanks :tada: Btw. I think you face this issue on Linux?

**Comment by shalithasuranga** [comment]
Hey @bmcgonag I tested this scenario by creating a tray menu and dynamically updating (updated several times -- not just one time) it by calling `setTray`. It worked fine on Linux (Ubuntu 20.4). Could you please post a screen recording if possilbe, then we can investigate the issue further. Thanks :tada:

**Reply by bmcgonag** [reply]
Sorry it took a couple of days. Been out of town. Here is a link to a shared video from my NextCloud. Let me know if you need anything else.

https://mync.routemehome.org/s/TgfqX2s4aNBbYWx
